### DISCUSSION TITLE
Incorrect permission (permission denied) on tfvars file

### DISCUSSION THREAD

**Discussion by ghost** [chadgeary/cloudblock#12] [Q&A] [answered]
I am getting an error when applying terraform

Command > ~/cloudblock/aws$ sudo terraform apply -var-file="aws.tfvars"

Error: error configuring Terraform AWS Provider: no valid credential sources for Terraform AWS Provider found.

Please see https://registry.terraform.io/providers/hashicorp/aws
for more information about providing credentials.

Error: NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors

**Comment by chadgeary** [comment, marked as answer]
ANSWER:
`sudo chown $USER *.tfvars && sudo chmod 640 *.tfvars`

My hunch is it has something to do with either:
* elevated powershell for WSL when it shouldn't be elevated, or
* the file is left open in explorer (save but not quit)

I have not been able to re-produce it.


###

Hey @mnavidh - sudo should not be used here.

From the README:
```
# In powershell's WSL window, change to the project's aws directory
cd ~/cloudblock/aws/

# Initialize terraform and apply the terraform state
terraform init
terraform apply -var-file="aws.tfvars"

# Note the outputs from terraform after the apply completes

# Wait for the virtual machine to become ready (Ansible will setup the services for us)
```

Regards

**Reply by ghost** [reply]
I had to use sudo because when I ran the command, I got a permission denied error for the variables file

**Reply by chadgeary** [reply]
Oh - were you the one that chatted in discord? For those that may come across this as well (if permissions on the vars file is wrong):
`sudo chown $USER *.tfvars && sudo chmod 640 *.tfvars`

My hunch is it has something to do with either A. elevated powershell for WSL when it shouldn't be elevated or B. the file is left open in explorer (save but not quit)

I have not been able to re-produce it.

 Cheers!

**Reply by ghost** [reply]
That worked! Thank you. Setting the permissions explicitly seems to be the one. I was not on the discord chat. I had the suspicion about file left open so I even did a restart of my computer and it did not work.

**Reply by chadgeary** [reply]
Glad to hear it is working - great!

👍

**Comment by chadgeary** [comment]
Marking as answered - though if someone can reproduce I'd like to update the README / notes.
